src/unittest/python/test_thresholding_manager.py

```diff
@@ -0,0 +1,169 @@
+"""
+----------------------------------------------------------------------
+Authors: Stephan Nell
+----------------------------------------------------------------------
+Unit tests for the Thresholding Manager
+----------------------------------------------------------------------
+"""
+import pytest
+import numpy as np
+from image_preprocessing.thresholding_manager import ThresholdingManager
+
+# blank image to test with Height set 1 Width set at 1
+blank_image = np.zeros((1, 1), dtype=np.uint8)
+
+
+def test_thresholding_constructor_incorrect_type():
+    """
+    Test to see if constructor check for invalid type i.e not String type
+    """
+    with pytest.raises(TypeError):
+        ThresholdingManager(123)
+
+
+def test_thresholding_constructor_paramter_number():
+    """
+    Tests with incorrect number of parameters
+    """
+    with pytest.raises(TypeError):
+        ThresholdingManager("adaptive", 1)
+
+
+def test_thresholding_type():
+    """
+    Tests the verify function with a invalid Thresholding type
+    """
+    with pytest.raises(NameError):
+        ThresholdingManager("Incorrect Thresholding type")
+
+
+def test_thresholding_type_2():
+    """
+    Tests for adaptive thresholding passed to constructor
+    """
+    ThresholdingManager("adaptive")
+
+
+def test_thresholding_type_3():
+    """
+    Tests for otsu thresholding passed to constructor
+    """
+    ThresholdingManager("otsu")
+
+
+def test_thresholding_type_4():
+    """
+    Tests for missing parameter
+    """
+    with pytest.raises(TypeError):
+        ThresholdingManager()
+
+
+def test_thresholding_type_5():
+    """
+    Tests for Adaptive thresholding type passed as capital letter
+    """
+    ThresholdingManager("Adaptive")
+
+
+def test_thresholding_type_6():
+    """
+    Tests for Otsu thresholding type passed as capital letter
+    """
+    ThresholdingManager("Otsu")
+
+
+def test_apply():
+    """
+    Tests for apply function with Adaptive type
+    """
+    manager = ThresholdingManager("adaptive")
```

Comment:
Check if the correct value is actually used.

Reply:
Once again these functions return very large numpy arrays. So these tests are merely for coverage and to check for basic failures.

At the moment the only solution I can think of is to insert `cv2.adaptiveThreshold(image, 255, cv2.ADAPTIVE_THRESH_GAUSSIAN_C, cv2.THRESH_BINARY_INV, 25, 10)` == the tested image

But this sorta defeats the purpose of the test since we will just be rewriting the function in the test